### ISSUE TITLE
Unconnectable Testnet3 providers. could not send p2p session create request: timeout waiting for reply to "p2p-session-create": p2p send timeout

### ISSUE THREAD

**Issue by Waldz** [mysteriumnetwork/node#3833]
**Describe the bug**
A significant amount of providers in Testnet3 version are unconnectible.

**To Reproduce**
1. Go to hosting droplet
2. Launch `myst` consumer inside Docker container
3. Connect to mentioned provider  with`myst cli connect 0x.. 0x.. wireguard`
4. See an error

**Environment (please complete the following information):**
 - Node version:
```
docker exec -ti supernode-tunnel-0 myst version
Mysterium Node
  Version: 0.60.0
```
 - OS:
 ```
uname -a
Linux menulis 4.19.0-16-amd64 #1 SMP Debian 4.19.181-1 (2021-03-19) x86_64 GNU/Linux

docker exec -ti supernode-tunnel-0 cat /etc/alpine-release
3.12.7
```
 - Docker (if applicable): specify docker version and image tag
 ```
docker version
Client: Docker Engine - Community
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        f0df350
 Built:             Wed Jun  2 11:56:47 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       b0f5bc3
  Built:            Wed Jun  2 11:54:58 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.6
  GitCommit:        d71fcd7d8303cbf684402823e425e9dd2e99285d
 runc:
  Version:          1.0.0-rc95
  GitCommit:        b9ee9c6314599f1b4a7f497e1f1f856fe433d3b7
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
 ```

**Comment by Waldz** [comment]
Provider: 0x57daba353514f004c86edf2e3892740ec4d98c4d
Problem: could not send p2p session create request: timeout waiting for reply to "p2p-session-create": p2p send timeout

Logs:
```
2021-08-19 09:33:29 [ERROR] [0x57daba353514f004c86edf2e3892740ec4d98c4d:supernode-tunnel-0] Connection error. server response invalid: 500 Internal Server Error (http://supernode-tunnel-0:4050/connection). Possible error: could not send p2p session create request: timeout waiting for reply to "p2p-session-create": p2p send timeout
```

Consumer logs:
```
2021-08-19T09:32:47.434 DBG market/mysterium/mysterium_api.go:66     > Total proposals: 1 supported: 1
2021-08-19T09:32:47.436 DBG config/config.go:224                     > Returning default value chain-id:80001
2021-08-19T09:32:47.437 DBG config/config.go:224                     > Returning default value payments.consumer.offchain-expiration:30m0s
2021-08-19T09:32:47.440 INF core/connection/manager.go:671           > Connection state: NotConnected -> Connecting
2021-08-19T09:32:47.441 DBG eventbus/event_bus.go:81                 > Published topic="State" event={State:Connecting SessionInfo:{StartedAt:2021-08-19 09:32:47.440265542 +0000 UTC m=+66.519443004 ConsumerID:{Address:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb} ConsumerLocation:{IP:168.119.238.125 ASN:24940 ISP:Hetzner Online GmbH Continent:EU Country:DE City:Unknown IPType:hosting} HermesID:0x7119442C7E627438deb0ec59291e31378F88DD06 State:Connecting SessionID: Proposal:{ServiceProposal:{ID:0 Format:service-proposal/v3 Compatibility:0 ProviderID:0x57daba353514f004c86edf2e3892740ec4d98c4d ServiceType:wireguard Location:{Continent:EU Country:GB City:Scotland ASN:2856 ISP:BT IPType:residential} Contacts:[{Type:nats/p2p/v1 Definition:{BrokerAddresses:[nats://testnet3-broker.mysterium.network:4222 nats://testnet3-broker.mysterium.network:4222 nats://167.233.11.60:4222]}}] AccessPolicies:<nil> Quality:{Quality:2.7 Latency:0 Bandwidth:0}} Price:88575017478116/h, 88575017478116816/GiB }}}
2021-08-19T09:32:47.443 DBG config/config.go:224                     > Returning default value default-currency:MYSTT
2021-08-19T09:32:47.444 INF core/state/state.go:439                  > Session ID: , state: Connecting, duration: 4.058336ms data: 37.8KiB/46.9KiB, throughput: 0bs/255Bs, spent: 0.000000MYSTT
2021-08-19T09:32:47.445 DBG communication/nats/connector.go:78       > Connecting to NATS servers: [nats://testnet3-broker.mysterium.network:4222 nats://testnet3-broker.mysterium.network:4222 nats://167.233.11.60:4222]
2021-08-19T09:32:47.446 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://testnet3-broker.mysterium.network:4222 access
2021-08-19T09:32:47.446 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://testnet3-broker.mysterium.network:4222 access
2021-08-19T09:32:47.447 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://167.233.11.60:4222 access
2021-08-19T09:32:47.447 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://testnet3-broker.mysterium.network:4222 access
2021-08-19T09:32:47.448 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://167.233.11.60:4222 access
2021-08-19T09:32:47.449 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://testnet3-broker.mysterium.network:4222 access
2021-08-19T09:32:47.449 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://167.233.11.60:4222 access
2021-08-19T09:32:47.450 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://167.233.11.60:4222 access
2021-08-19T09:32:48.958 DBG p2p/dialer.go:184                        > Consumer 0xa53a9a07970ab2fdc04217e2d4a807bd350203eb sending public key 58dff2f666bc0c6779e92daa2a8678c083738491a8ffc6b70e8801af65a64340 to provider 0x57daba353514f004c86edf2e3892740ec4d98c4d
2021-08-19T09:32:49.398 DBG p2p/dialer.go:211                        > Consumer 0xa53a9a07970ab2fdc04217e2d4a807bd350203eb received provider 0x57daba353514f004c86edf2e3892740ec4d98c4d with config: publicIP:"86.134.59.214"  ports:52042  ports:52049
2021-08-19T09:32:49.400 DBG core/ip/cached_resolver.go:75            > Found cached public IP
2021-08-19T09:32:49.401 INF core/port/pool.go:59                     > Supplying port 34798
2021-08-19T09:32:49.403 INF core/port/pool.go:59                     > Supplying port 35679
2021-08-19T09:32:49.406 DBG config/config.go:224                     > Returning default value stun-servers:[stun.l.google.com:19302 stun1.l.google.com:19302 stun2.l.google.com:19302]
2021-08-19T09:32:49.449 DBG eventbus/event_bus.go:81                 > Published topic="STUN detection" event={Identity:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb NATType:full}
2021-08-19T09:32:49.451 DBG eventbus/event_bus.go:81                 > Published topic="STUN detection" event={Identity:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb NATType:full}
2021-08-19T09:32:49.454 DBG p2p/dialer.go:237                        > Consumer 0xa53a9a07970ab2fdc04217e2d4a807bd350203eb sending ack with encrypted config to provider 0x57daba353514f004c86edf2e3892740ec4d98c4d
2021-08-19T09:32:49.528 INF firewall/outgoing_firewall_noop.go:48    > Allow IP 86.134.59.214 access
2021-08-19T09:32:49.529 DBG p2p/dialer.go:282                        > Skipping provider ping
2021-08-19T09:32:49.539 DBG p2p/dialer.go:133                        > Received handlers ready message from provider
2021-08-19T09:32:49.542 DBG p2p/channel.go:212                       > Creating p2p channel with local addr: 172.17.0.3:34798, UDP session addr: 127.0.0.1:57484, proxy addr: 127.0.0.1:50310, remote peer addr: x.x.x.x:52042
2021-08-19T09:32:49.544 DBG p2p/channel.go:555                       > Will use service conn with local port: 35679, remote port: 52049
2021-08-19T09:32:49.545 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://testnet3-broker.mysterium.network:4222 removed
2021-08-19T09:32:49.546 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://testnet3-broker.mysterium.network:4222 removed
2021-08-19T09:32:49.547 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://167.233.11.60:4222 removed
2021-08-19T09:32:49.549 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://testnet3-broker.mysterium.network:4222 removed
2021-08-19T09:32:49.549 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://167.233.11.60:4222 removed
2021-08-19T09:32:49.550 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://testnet3-broker.mysterium.network:4222 removed
2021-08-19T09:32:49.551 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://167.233.11.60:4222 removed
2021-08-19T09:32:49.552 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://167.233.11.60:4222 removed
2021-08-19T09:32:49.553 DBG config/config.go:224                     > Returning default value chain-id:80001
2021-08-19T09:32:49.554 DBG core/connection/manager.go:540           > Sending P2P message to "p2p-session-create": consumer:{id:"0xa53a9a07970ab2fdc04217e2d4a807bd350203eb"  hermesID:"0x7119442C7E627438deb0ec59291e31378F88DD06"  paymentVersion:"v3"  location:{country:"DE"}  pricing:{PerGib:"\x01:\xae\x82T\xc1\xd5\xd0"  PerHour:"P\x8e\xfas\xe7\xe4"}}  config:"{\"PublicKey\":\"UiXuu31xX9US//b3B9M8VO1q02v5LAo3DUIoIkIoTWE=\",\"Ports\":null}"
2021-08-19T09:32:49.549 WRN communication/nats/connection_wrap.go:102 > NATS: disconnected
2021-08-19T09:32:49.556 WRN communication/nats/connection_wrap.go:101 > NATS: connection closed
2021-08-19T09:32:49.556 DBG session/pingpong/invoice_payer.go:127    > Starting...
2021-08-19T09:33:09.556 DBG core/connection/manager.go:391           > Sending session status P2P message to "p2p-session-connectivity-status": ConsumerID:"0xa53a9a07970ab2fdc04217e2d4a807bd350203eb"  Code:2000  Message:"could not send p2p session create request: timeout waiting for reply to \"p2p-session-create\": p2p send timeout"
2021-08-19T09:33:14.377 WRN communication/nats/connection_wrap.go:102 > NATS: disconnected
2021-08-19T09:33:21.887 WRN communication/nats/connection_wrap.go:103 > NATS: reconnected
2021-08-19T09:33:29.561 ERR core/connection/manager.go:231           > Connect failed, disconnecting error="could not send p2p session create request: timeout waiting for reply to \"p2p-session-create\": p2p send timeout"
2021-08-19T09:33:29.563 DBG session/pingpong/invoice_payer.go:296    > Stopping...
2021-08-19T09:33:29.564 DBG core/ip/cached_resolver.go:91            > Clearing ip resolver cache
2021-08-19T09:33:29.566 INF core/connection/manager.go:671           > Connection state: Connecting -> NotConnected
2021-08-19T09:33:29.568 DBG eventbus/event_bus.go:81                 > Published topic="State" event={State:NotConnected SessionInfo:{StartedAt:2021-08-19 09:32:47.440265542 +0000 UTC m=+66.519443004 ConsumerID:{Address:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb} ConsumerLocation:{IP:168.119.238.125 ASN:24940 ISP:Hetzner Online GmbH Continent:EU Country:DE City:Unknown IPType:hosting} HermesID:0x7119442C7E627438deb0ec59291e31378F88DD06 State:NotConnected SessionID: Proposal:{ServiceProposal:{ID:0 Format:service-proposal/v3 Compatibility:0 ProviderID:0x57daba353514f004c86edf2e3892740ec4d98c4d ServiceType:wireguard Location:{Continent:EU Country:GB City:Scotland ASN:2856 ISP:BT IPType:residential} Contacts:[{Type:nats/p2p/v1 Definition:{BrokerAddresses:[nats://testnet3-broker.mysterium.network:4222 nats://testnet3-broker.mysterium.network:4222 nats://167.233.11.60:4222]}}] AccessPolicies:<nil> Quality:{Quality:2.7 Latency:0 Bandwidth:0}} Price:88575017478116/h, 88575017478116816/GiB }}}
2021-08-19T09:33:29.570 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID: Key:Consumer whole Connect Duration:42.133905923s}
2021-08-19T09:33:29.571 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID: Key:Consumer P2P channel creation Duration:2.110104312s}
2021-08-19T09:33:29.572 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID: Key:Consumer P2P connect Duration:1.512781845s}
2021-08-19T09:33:29.573 DBG core/location/oracle_resolver.go:64      > Detecting with oracle resolver
2021-08-19T09:33:29.575 DBG config/config.go:224                     > Returning default value default-currency:MYSTT
2021-08-19T09:33:29.576 INF core/state/state.go:439                  > Session ID: , state: NotConnected, duration: 42.13573087s data: 0b/0b, throughput: 0bs/0bs, spent: 0.000000MYSTT
2021-08-19T09:33:29.577 WRN core/quality/sender.go:360               > Can't recover session context error="unknown session: "
2021-08-19T09:33:29.578 WRN core/quality/sender.go:360               > Can't recover session context error="unknown session: "
2021-08-19T09:33:29.578 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID: Key:Consumer P2P exchange Duration:441.911174ms}
2021-08-19T09:33:29.579 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID: Key:Consumer P2P exchange (ports) Duration:3.932847ms}
2021-08-19T09:33:29.579 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID: Key:Consumer P2P exchange ack Duration:74.872173ms}
2021-08-19T09:33:29.580 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID: Key:Consumer P2P dial (upnp) Duration:4.030237ms}
2021-08-19T09:33:29.580 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID: Key:Consumer P2P dial ack Duration:12.558771ms}
2021-08-19T09:33:29.581 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID: Key:Consumer session creation Duration:20.00212949s}
2021-08-19T09:33:29.581 DBG core/connection/manager.go:203           > Consumer connection trace: "Consumer whole Connect" took 42.133905923s, "Consumer P2P channel creation" took 2.110104312s, "Consumer P2P connect" took 1.512781845s, "Consumer P2P exchange" took 441.911174ms, "Consumer P2P exchange (ports)" took 3.932847ms, "Consumer P2P exchange ack" took 74.872173ms, "Consumer P2P dial (upnp)" took 4.030237ms, "Consumer P2P dial ack" took 12.558771ms, "Consumer session creation" took 20.00212949s
2021-08-19T09:33:29.582 DBG eventbus/event_bus.go:81                 > Published topic="connection_events" event={ServiceType:wireguard ProviderID:0x57daba353514f004c86edf2e3892740ec4d98c4d ConsumerID:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb HermesID:0x7119442C7E627438deb0ec59291e31378F88DD06 Error:could not send p2p session create request: timeout waiting for reply to "p2p-session-create": p2p send timeout Stage:connection_unknown_error}
2021-08-19T09:33:29.583 ERR tequilapi/endpoints/connection.go:221    > Failed to connect error="could not send p2p session create request: timeout waiting for reply to \"p2p-session-create\": p2p send timeout"
2021-08-19T09:33:29.588 WRN core/quality/sender.go:360               > Can't recover session context error="unknown session: "
2021-08-19T09:33:29.589 WRN core/quality/sender.go:360               > Can't recover session context error="unknown session: "
2021-08-19T09:33:29.589 WRN core/quality/sender.go:360               > Can't recover session context error="unknown session: "
2021-08-19T09:33:29.590 WRN core/quality/sender.go:360               > Can't recover session context error="unknown session: "
2021-08-19T09:33:29.590 WRN core/quality/sender.go:360               > Can't recover session context error="unknown session: "
2021-08-19T09:33:29.590 WRN core/quality/sender.go:360               > Can't recover session context error="unknown session: "
2021-08-19T09:33:29.591 WRN core/quality/sender.go:360               > Can't recover session context error="unknown session: "
2021-08-19T09:33:29.593 DBG eventbus/event_bus.go:81                 > Published topic="location-update-event" event={IP:168.119.238.125 ASN:24940 ISP:Hetzner Online GmbH Continent:EU Country:DE City:Unknown IPType:hosting}
2021-08-19T09:33:29.593 DBG core/location/cache.go:106               > Location update succeeded: {168.119.238.125 24940 Hetzner Online GmbH EU DE Unknown hosting}
```

**Comment by Waldz** [comment]
Provider: 0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21
Problem: Location check lasts 20s and fails: `server response invalid: 503 Service Unavailable (http://supernode-tunnel-1:4050/connection/location). Possible error: failed to execute request: Get https://testnet3-location.mysterium.network/api/v1/location: net/http: request canceled (Client.Timeout exceeded while awaiting headers)`

Logs:
```
2021-08-19 10:02:06 [INFO] [0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21:supernode-tunnel-0] Control connected: version: 0.60.0 uptime: 8m25.581571724s
2021-08-19 10:02:06 [INFO] [0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21:supernode-tunnel-0] Control ok - unlocking identity...
2021-08-19 10:02:06 [INFO] [0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21:supernode-tunnel-0] Consumer identity ready: {0xa53a9a07970ab2fdc04217e2d4a807bd350203eb}
2021-08-19 10:02:06 [INFO] [0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21:supernode-tunnel-0] Unlocked identity 0xa53a9a07970ab2fdc04217e2d4a807bd350203eb - creating connection...
2021-08-19 10:02:06 [INFO] [0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21:supernode-tunnel-0] Initial ip: 168.119.238.125
2021-08-19 10:02:11 [WARN] Failed to get provider: all providers already taken
2021-08-19 10:02:11 [INFO] Waiting 5 seconds before next try.
2021-08-19 10:02:16 [WARN] Failed to get provider: all providers already taken
2021-08-19 10:02:16 [INFO] Waiting 5 seconds before next try.
[GIN] 2021/08/19 - 10:02:16 | 200 |     3.13954ms |      172.18.0.4 | GET      "/metrics"
2021-08-19 10:02:21 [WARN] Failed to get provider: all providers already taken
2021-08-19 10:02:21 [INFO] Waiting 5 seconds before next try.
2021-08-19 10:02:26 [WARN] Failed to get provider: all providers already taken
2021-08-19 10:02:26 [INFO] Waiting 5 seconds before next try.
2021-08-19 10:02:31 [WARN] Failed to get provider: all providers already taken
2021-08-19 10:02:31 [INFO] Waiting 5 seconds before next try.
2021-08-19 10:02:36 [WARN] Failed to get provider: all providers already taken
2021-08-19 10:02:36 [INFO] Waiting 5 seconds before next try.
2021-08-19 10:02:41 [WARN] Failed to get provider: all providers already taken
2021-08-19 10:02:41 [INFO] Waiting 5 seconds before next try.
2021-08-19 10:02:46 [WARN] Failed to get provider: all providers already taken
2021-08-19 10:02:46 [INFO] Waiting 5 seconds before next try.
[GIN] 2021/08/19 - 10:02:46 | 200 |    2.988133ms |      172.18.0.4 | GET      "/metrics"
2021-08-19 10:02:51 [WARN] Failed to get provider: all providers already taken
2021-08-19 10:02:51 [INFO] Waiting 5 seconds before next try.
{"level":"error","error":"server response invalid: 503 Service Unavailable (http://supernode-tunnel-0:4050/connection/location). Possible error: failed to execute request: Get https://testnet3-location.mysterium.network/api/v1/location: net/http: request canceled (Client.Timeout exceeded while awaiting headers)","time":"2021-08-19T10:02:56Z"}
2021-08-19 10:02:56 [WARN] [0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21:supernode-tunnel-0] Location check failed. server response invalid: 503 Service Unavailable (http://supernode-tunnel-0:4050/connection/location). Possible error: failed to execute request: Get https://testnet3-location.mysterium.network/api/v1/location: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
2021-08-19 10:02:56 [INFO] [0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21:supernode-tunnel-0] Connection established to: 0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21 Session-> e2113530-d05f-4f5d-95aa-d99100e33606
```

Consumer logs:
```
2021-08-19T10:02:06.885 DBG identity/selector/handler.go:98          > Attempting to use last identity
2021-08-19T10:02:06.887 DBG identity/selector/handler.go:103         > Found identity in cache: 0xa53a9a07970ab2fdc04217e2d4a807bd350203eb
2021-08-19T10:02:06.888 DBG identity/manager.go:153                  > Unlocked identity found in cache, skipping keystore: 0xa53a9a07970ab2fdc04217e2d4a807bd350203eb
2021-08-19T10:02:06.888 DBG identity/selector/handler.go:108         > Unlocked identity: 0xa53a9a07970ab2fdc04217e2d4a807bd350203eb
2021-08-19T10:02:06.889 DBG config/config.go:224                     > Returning default value chain-id:80001
2021-08-19T10:02:06.893 DBG core/ip/cached_resolver.go:79            > Public IP cache is empty, fetching IP
2021-08-19T10:02:06.899 DBG core/ip/resolver.go:105                  > IP detected: 168.119.238.125
2021-08-19T10:02:06.901 DBG config/config.go:224                     > Returning default value chain-id:80001
2021-08-19T10:02:06.902 DBG config/config.go:224                     > Returning default value chain-id:80001
2021-08-19T10:02:06.902 DBG eventbus/event_bus.go:81                 > Published topic="connection_events" event={ServiceType:wireguard ProviderID:0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21 ConsumerID:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb HermesID:0x7119442C7E627438deb0ec59291e31378F88DD06 Error: Stage:registration_registered}
2021-08-19T10:02:06.903 INF tequilapi/endpoints/connection.go:184    > identity "0xa53a9a07970ab2fdc04217e2d4a807bd350203eb" is registered, continuing...
2021-08-19T10:02:07.218 DBG market/mysterium/mysterium_api.go:66     > Total proposals: 1 supported: 1
2021-08-19T10:02:07.220 DBG config/config.go:224                     > Returning default value chain-id:80001
2021-08-19T10:02:07.222 DBG config/config.go:224                     > Returning default value payments.consumer.offchain-expiration:30m0s
2021-08-19T10:02:07.223 INF core/connection/manager.go:671           > Connection state: NotConnected -> Connecting
2021-08-19T10:02:07.225 DBG eventbus/event_bus.go:81                 > Published topic="State" event={State:Connecting SessionInfo:{StartedAt:2021-08-19 10:02:07.223492133 +0000 UTC m=+508.119354628 ConsumerID:{Address:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb} ConsumerLocation:{IP:168.119.238.125 ASN:24940 ISP:Hetzner Online GmbH Continent:EU Country:DE City:Unknown IPType:hosting} HermesID:0x7119442C7E627438deb0ec59291e31378F88DD06 State:Connecting SessionID: Proposal:{ServiceProposal:{ID:0 Format:service-proposal/v3 Compatibility:0 ProviderID:0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21 ServiceType:wireguard Location:{Continent:EU Country:GB City:England ASN:13285 ISP:TalkTalk IPType:residential} Contacts:[{Type:nats/p2p/v1 Definition:{BrokerAddresses:[nats://testnet3-broker.mysterium.network:4222 nats://testnet3-broker.mysterium.network:4222 nats://167.233.11.60:4222]}}] AccessPolicies:0xc000961500 Quality:{Quality:2.2 Latency:0 Bandwidth:0}} Price:111038981468210/h, 111038981468210864/GiB }}}
2021-08-19T10:02:07.227 DBG communication/nats/connector.go:78       > Connecting to NATS servers: [nats://testnet3-broker.mysterium.network:4222 nats://testnet3-broker.mysterium.network:4222 nats://167.233.11.60:4222]
2021-08-19T10:02:07.229 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://testnet3-broker.mysterium.network:4222 access
2021-08-19T10:02:07.231 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://testnet3-broker.mysterium.network:4222 access
2021-08-19T10:02:07.232 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://167.233.11.60:4222 access
2021-08-19T10:02:07.233 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://testnet3-broker.mysterium.network:4222 access
2021-08-19T10:02:07.234 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://167.233.11.60:4222 access
2021-08-19T10:02:07.235 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://testnet3-broker.mysterium.network:4222 access
2021-08-19T10:02:07.235 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://167.233.11.60:4222 access
2021-08-19T10:02:07.236 INF firewall/outgoing_firewall_noop.go:57    > Allow URL nats://167.233.11.60:4222 access
2021-08-19T10:02:07.231 DBG config/config.go:224                     > Returning default value default-currency:MYSTT
2021-08-19T10:02:07.239 INF core/state/state.go:439                  > Session ID: , state: Connecting, duration: 15.582919ms data: 15.3KiB/57.0KiB, throughput: 95Bs/95Bs, spent: 0.000000MYSTT
2021-08-19T10:02:08.754 DBG p2p/dialer.go:184                        > Consumer 0xa53a9a07970ab2fdc04217e2d4a807bd350203eb sending public key f820b20cf0d9679d3a02308ee8bc3de7293b052db15945abd7dc449447f2f436 to provider 0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21
2021-08-19T10:02:14.992 DBG p2p/dialer.go:211                        > Consumer 0xa53a9a07970ab2fdc04217e2d4a807bd350203eb received provider 0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21 with config: publicIP:"2.100.160.91"  ports:58558  ports:43595
2021-08-19T10:02:14.998 DBG core/ip/cached_resolver.go:75            > Found cached public IP
2021-08-19T10:02:14.999 INF core/port/pool.go:59                     > Supplying port 11564
2021-08-19T10:02:15.002 INF core/port/pool.go:59                     > Supplying port 48912
2021-08-19T10:02:15.003 DBG config/config.go:224                     > Returning default value stun-servers:[stun.l.google.com:19302 stun1.l.google.com:19302 stun2.l.google.com:19302]
2021-08-19T10:02:15.042 DBG eventbus/event_bus.go:81                 > Published topic="STUN detection" event={Identity:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb NATType:full}
2021-08-19T10:02:15.045 DBG eventbus/event_bus.go:81                 > Published topic="STUN detection" event={Identity:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb NATType:full}
2021-08-19T10:02:15.046 DBG p2p/dialer.go:237                        > Consumer 0xa53a9a07970ab2fdc04217e2d4a807bd350203eb sending ack with encrypted config to provider 0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21
2021-08-19T10:02:15.097 INF firewall/outgoing_firewall_noop.go:48    > Allow IP 2.100.160.91 access
2021-08-19T10:02:15.098 DBG p2p/dialer.go:282                        > Skipping provider ping
2021-08-19T10:02:15.100 DBG p2p/dialer.go:133                        > Received handlers ready message from provider
2021-08-19T10:02:15.103 DBG p2p/channel.go:212                       > Creating p2p channel with local addr: 172.17.0.3:11564, UDP session addr: 127.0.0.1:33420, proxy addr: 127.0.0.1:42799, remote peer addr: x.x.x.x:58558
2021-08-19T10:02:15.105 DBG p2p/channel.go:555                       > Will use service conn with local port: 48912, remote port: 43595
2021-08-19T10:02:15.106 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://testnet3-broker.mysterium.network:4222 removed
2021-08-19T10:02:15.107 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://testnet3-broker.mysterium.network:4222 removed
2021-08-19T10:02:15.107 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://167.233.11.60:4222 removed
2021-08-19T10:02:15.108 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://testnet3-broker.mysterium.network:4222 removed
2021-08-19T10:02:15.109 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://167.233.11.60:4222 removed
2021-08-19T10:02:15.109 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://testnet3-broker.mysterium.network:4222 removed
2021-08-19T10:02:15.110 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://167.233.11.60:4222 removed
2021-08-19T10:02:15.111 INF firewall/outgoing_firewall_noop.go:61    > Rule for URL: nats://167.233.11.60:4222 removed
2021-08-19T10:02:15.112 DBG config/config.go:224                     > Returning default value chain-id:80001
2021-08-19T10:02:15.113 DBG core/connection/manager.go:540           > Sending P2P message to "p2p-session-create": consumer:{id:"0xa53a9a07970ab2fdc04217e2d4a807bd350203eb"  hermesID:"0x7119442C7E627438deb0ec59291e31378F88DD06"  paymentVersion:"v3"  location:{country:"DE"}  pricing:{PerGib:"\x01\x8a}]\x86\xa2F\xb0"  PerHour:"d\xfdG\x0b\xf02"}}  config:"{\"PublicKey\":\"5k6gv5XA7E3Q7huDj1XwbXoEX3wy2B4UcGzjRLwYIjI=\",\"Ports\":null}"
2021-08-19T10:02:15.114 DBG session/pingpong/invoice_payer.go:127    > Starting...
2021-08-19T10:02:15.115 WRN communication/nats/connection_wrap.go:102 > NATS: disconnected
2021-08-19T10:02:15.116 WRN communication/nats/connection_wrap.go:101 > NATS: connection closed
2021-08-19T10:02:15.163 DBG session/pingpong/factory.go:136          > Received P2P message for "p2p-payment-invoice": AgreementID:"60933087532070207597448730368511591419058256564880633397550459210214220949217"  AgreementTotal:"1"  TransactorFee:"0"  Hashlock:"42dc95923806929c2a1f3c8a37d920708d0d21b5b37a0dfaf3bf17844577551e"  Provider:"0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21"  ChainID:80001
2021-08-19T10:02:15.166 DBG session/pingpong/invoice_payer.go:146    > Invoice received: {60933087532070207597448730368511591419058256564880633397550459210214220949217 1 0 42dc95923806929c2a1f3c8a37d920708d0d21b5b37a0dfaf3bf17844577551e 0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21 80001}
2021-08-19T10:02:15.167 DBG session/pingpong/price_calculator.go:51  > Calculated price 2168731728377666. Time component: 1.6215766733420157748e+09, data component: 2.1687301068009934375e+15
2021-08-19T10:02:15.169 DBG session/pingpong/invoice_payer.go:191    > Estimated tolerance 3, upper bound 6506195185132998
2021-08-19T10:02:15.170 DBG session/pingpong/invoice_payer.go:244    > Loaded previous state: already promised: 5360043187212549
2021-08-19T10:02:15.171 DBG session/pingpong/invoice_payer.go:245    > Incrementing promised amount by 1
2021-08-19T10:02:15.175 DBG session/pingpong/exchange_messaging.go:68 > Sending P2P message to "p2p-payment-message": Promise:{ChannelID:"\\`\x14I\xa3~\xf8\xf8\xb5D\xe2\x04\x8e<:\x93A\xa4#\x12"  Amount:"5360043187212550"  Fee:"0"  Hashlock:"Bܕ\x928\x06\x92\x9c*\x1f<\x8a7\xd9 p\x8d\r!\xb5\xb3z\r\xfa\xf3\xbf\x17\x84EwU\x1e"  ChainID:80001  Signature:"&Y\xe86\xd0~J0\xfc\xce.\x96\x00Ĵ9L\x012\x84\xe2\xca1W=\x1a=6\x07\x9ad\xc1.zt\xc0\xc7\x118>5\xa0\xe9\x14\xb4}\xd5\xf6\xfb\xbfYo1\x03\x0bП&-\xa6A\xd8<R\x1b"}  AgreementID:"60933087532070207597448730368511591419058256564880633397550459210214220949217"  AgreementTotal:"1"  Provider:"0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21"  Signature:"781f8fa3e23f24cec5fb4e6b04097cdc31fd1b5a327fd06fcdd839f54fad49717edef1fce22a31e6a3c7ac1e2a7dc055c01be4e9e011de718ef0a149e93fbbed1c"  HermesID:"0x7119442C7E627438deb0ec59291e31378F88DD06"  ChainID:80001
2021-08-19T10:02:15.210 DBG eventbus/event_bus.go:81                 > Published topic="consumer_grand_total_change" event={Current:+5360043187212550 ChainID:80001 HermesID:0x7119442C7E627438deb0ec59291e31378F88DD06 ConsumerID:{Address:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb}}
2021-08-19T10:02:15.211 DBG config/config.go:224                     > Returning default value payments.consumer.offchain-expiration:30m0s
2021-08-19T10:02:15.213 DBG eventbus/event_bus.go:81                 > Published topic="balance_change" event={Identity:{Address:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb} Previous:+99999976724922872496 Current:+99999935075076542208}
2021-08-19T10:02:16.496 INF core/connection/manager.go:553           > Provider's session config: {"local_port":0,"remote_port":0,"ports":null,"provider":{"public_key":"D6LSrg510qorRmrZrDLXW5HZhQnSXMjgwDnsBtATBEE=","endpoint":"2.100.160.91:43595"},"consumer":{"ip_address":"10.182.0.2/24","dns_ips":"10.182.0.1"}}
2021-08-19T10:02:16.499 DBG eventbus/event_bus.go:81                 > Published topic="Session" event={Status:Created SessionInfo:{StartedAt:2021-08-19 10:02:07.223492133 +0000 UTC m=+508.119354628 ConsumerID:{Address:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb} ConsumerLocation:{IP:168.119.238.125 ASN:24940 ISP:Hetzner Online GmbH Continent:EU Country:DE City:Unknown IPType:hosting} HermesID:0x7119442C7E627438deb0ec59291e31378F88DD06 State:Connecting SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Proposal:{ServiceProposal:{ID:0 Format:service-proposal/v3 Compatibility:0 ProviderID:0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21 ServiceType:wireguard Location:{Continent:EU Country:GB City:England ASN:13285 ISP:TalkTalk IPType:residential} Contacts:[{Type:nats/p2p/v1 Definition:{BrokerAddresses:[nats://testnet3-broker.mysterium.network:4222 nats://testnet3-broker.mysterium.network:4222 nats://167.233.11.60:4222]}}] AccessPolicies:0xc000961500 Quality:{Quality:2.2 Latency:0 Bandwidth:0}} Price:111038981468210/h, 111038981468210864/GiB }}}
2021-08-19T10:02:16.504 DBG consumer/session/session_storage.go:330  > Session e2113530-d05f-4f5d-95aa-d99100e33606 saved
2021-08-19T10:02:16.506 DBG core/ip/cached_resolver.go:75            > Found cached public IP
2021-08-19T10:02:16.507 INF firewall/outgoing_firewall_noop.go:48    > Allow IP 2.100.160.91 access
2021-08-19T10:02:16.507 DBG core/connection/dns_option.go:85         > Selecting DNS servers using strategy: auto
2021-08-19T10:02:16.509 DBG core/connection/dns_option.go:95         > Attempting to use provider DNS
2021-08-19T10:02:16.510 INF services/wireguard/connection/connection.go:144 > Starting new connection
2021-08-19T10:02:16.511 DBG config/config.go:224                     > Returning default value usermode:false
2021-08-19T10:02:16.522 DBG services/wireguard/endpoint/wg_client.go:59 > "ip link add iswgsupported type wireguard" output:

2021-08-19T10:02:16.548 DBG services/wireguard/endpoint/wg_client.go:64 > "ip link del iswgsupported" output:

2021-08-19T10:02:16.551 INF services/wireguard/connection/connection.go:182 > Starting connection endpoint
2021-08-19T10:02:16.553 DBG services/wireguard/endpoint/endpoint.go:63 > Allocated interface: myst0
2021-08-19T10:02:16.562 DBG services/wireguard/endpoint/kernelspace/client.go:165 > "ip link add dev myst0 type wireguard" output:

2021-08-19T10:02:16.569 DBG services/wireguard/endpoint/kernelspace/client.go:173 > "ip address replace dev myst0 10.182.0.2/24" output:

2021-08-19T10:02:16.573 DBG services/wireguard/endpoint/kernelspace/client.go:178 > "ip link set dev myst0 up" output:

2021-08-19T10:02:16.585 DBG utils/netutil/network_linux.go:40        > "ip route add 2.100.160.91 via 172.17.0.1" output:

2021-08-19T10:02:16.593 DBG utils/netutil/network_linux.go:48        > "ip route add 0.0.0.0/1 dev myst0" output:

2021-08-19T10:02:16.599 DBG utils/netutil/network_linux.go:52        > "ip route add 128.0.0.0/1 dev myst0" output:

2021-08-19T10:02:16.602 DBG utils/netutil/network_linux.go:77        > "sysctl net.ipv6.conf.all.disable_ipv6" output:
net.ipv6.conf.all.disable_ipv6 = 0

2021-08-19T10:02:16.606 DBG utils/netutil/network_linux.go:57        > "ip -6 route add ::/1 dev myst0" output:

2021-08-19T10:02:16.610 DBG utils/netutil/network_linux.go:61        > "ip -6 route add 8000::/1 dev myst0" output:

2021-08-19T10:02:16.640 INF services/wireguard/connection/connection.go:165 > Adding connection peer 2.100.160.91:43595
2021-08-19T10:02:16.641 INF services/wireguard/connection/connection.go:167 > Waiting for initial handshake
2021-08-19T10:02:16.777 DBG core/ip/cached_resolver.go:59            > Outbound IP cache is empty, fetching IP
2021-08-19T10:02:16.779 INF firewall/outgoing_firewall_noop.go:40    > Outgoing traffic block requested
2021-08-19T10:02:16.780 DBG core/connection/manager.go:768           > waiting for connected state
2021-08-19T10:02:16.781 DBG core/connection/manager.go:800           > Connection state received: Connecting
2021-08-19T10:02:16.783 DBG core/connection/manager.go:778           > Connected started event received
2021-08-19T10:02:16.784 DBG core/connection/manager.go:800           > Connection state received: Connected
2021-08-19T10:02:16.785 INF core/connection/manager.go:671           > Connection state: Connecting -> Connected
2021-08-19T10:02:16.786 DBG eventbus/event_bus.go:81                 > Published topic="State" event={State:Connected SessionInfo:{StartedAt:2021-08-19 10:02:07.223492133 +0000 UTC m=+508.119354628 ConsumerID:{Address:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb} ConsumerLocation:{IP:168.119.238.125 ASN:24940 ISP:Hetzner Online GmbH Continent:EU Country:DE City:Unknown IPType:hosting} HermesID:0x7119442C7E627438deb0ec59291e31378F88DD06 State:Connected SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Proposal:{ServiceProposal:{ID:0 Format:service-proposal/v3 Compatibility:0 ProviderID:0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21 ServiceType:wireguard Location:{Continent:EU Country:GB City:England ASN:13285 ISP:TalkTalk IPType:residential} Contacts:[{Type:nats/p2p/v1 Definition:{BrokerAddresses:[nats://testnet3-broker.mysterium.network:4222 nats://testnet3-broker.mysterium.network:4222 nats://167.233.11.60:4222]}}] AccessPolicies:0xc000961500 Quality:{Quality:2.2 Latency:0 Bandwidth:0}} Price:111038981468210/h, 111038981468210864/GiB }}}
2021-08-19T10:02:16.788 DBG core/connection/manager.go:561           > Sending P2P message to "p2p-session-acknowledge": consumerID:"0xa53a9a07970ab2fdc04217e2d4a807bd350203eb"  sessionID:"e2113530-d05f-4f5d-95aa-d99100e33606"
2021-08-19T10:02:16.790 DBG core/location/oracle_resolver.go:64      > Detecting with oracle resolver
2021-08-19T10:02:16.792 DBG config/config.go:224                     > Returning default value default-currency:MYSTT
2021-08-19T10:02:16.792 INF core/state/state.go:439                  > Session ID: e2113530-d05f-4f5d-95aa-d99100e33606, state: Connected, duration: 9.569030524s data: 15.3KiB/57.0KiB, throughput: 95Bs/95Bs, spent: 0.000000MYSTT
2021-08-19T10:02:16.793 INF cmd/di.go:978                            > Reconnecting HTTP clients due to VPN connection state change
2021-08-19T10:02:16.794 DBG eventbus/event_bus.go:81                 > Published topic="ether-client-reconnect" event={}
2021-08-19T10:02:16.795 DBG identity/registry/registry_contract.go:315 > Loading initial state
2021-08-19T10:02:16.796 DBG identity/registry/registry_contract.go:330 > Identity {"0xa53a9a07970ab2fdc04217e2d4a807bd350203eb"} already registered, skipping
2021-08-19T10:02:16.797 DBG core/ip/cached_resolver.go:91            > Clearing ip resolver cache
2021-08-19T10:02:16.797 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID:e2113530-d05f-4f5d-95aa-d99100e33606 Key:Consumer whole Connect Duration:9.577141599s}
2021-08-19T10:02:16.798 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID:e2113530-d05f-4f5d-95aa-d99100e33606 Key:Consumer P2P channel creation Duration:7.884935464s}
2021-08-19T10:02:16.798 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID:e2113530-d05f-4f5d-95aa-d99100e33606 Key:Consumer P2P connect Duration:1.526747001s}
2021-08-19T10:02:16.799 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID:e2113530-d05f-4f5d-95aa-d99100e33606 Key:Consumer P2P exchange Duration:6.244404966s}
2021-08-19T10:02:16.800 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID:e2113530-d05f-4f5d-95aa-d99100e33606 Key:Consumer P2P exchange (ports) Duration:4.869286ms}
2021-08-19T10:02:16.800 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID:e2113530-d05f-4f5d-95aa-d99100e33606 Key:Consumer P2P exchange ack Duration:50.854286ms}
2021-08-19T10:02:16.801 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID:e2113530-d05f-4f5d-95aa-d99100e33606 Key:Consumer P2P dial (upnp) Duration:2.993177ms}
2021-08-19T10:02:16.801 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID:e2113530-d05f-4f5d-95aa-d99100e33606 Key:Consumer P2P dial ack Duration:5.88626ms}
2021-08-19T10:02:16.802 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID:e2113530-d05f-4f5d-95aa-d99100e33606 Key:Consumer session creation Duration:1.385978168s}
2021-08-19T10:02:16.802 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID:e2113530-d05f-4f5d-95aa-d99100e33606 Key:Consumer session creation (start) Duration:7.137642ms}
2021-08-19T10:02:16.802 DBG eventbus/event_bus.go:81                 > Published topic="Trace" event={ID:e2113530-d05f-4f5d-95aa-d99100e33606 Key:Consumer start connection Duration:291.453602ms}
2021-08-19T10:02:16.803 DBG core/connection/manager.go:203           > Consumer connection trace: "Consumer whole Connect" took 9.577141599s, "Consumer P2P channel creation" took 7.884935464s, "Consumer P2P connect" took 1.526747001s, "Consumer P2P exchange" took 6.244404966s, "Consumer P2P exchange (ports)" took 4.869286ms, "Consumer P2P exchange ack" took 50.854286ms, "Consumer P2P dial (upnp)" took 2.993177ms, "Consumer P2P dial ack" took 5.88626ms, "Consumer session creation" took 1.385978168s, "Consumer session creation (start)" took 7.137642ms, "Consumer start connection" took 291.453602ms
2021-08-19T10:02:16.803 DBG eventbus/event_bus.go:81                 > Published topic="connection_events" event={ServiceType:wireguard ProviderID:0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21 ConsumerID:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb HermesID:0x7119442C7E627438deb0ec59291e31378F88DD06 Error: Stage:connection_ok}
2021-08-19T10:02:16.805 DBG core/ip/cached_resolver.go:79            > Public IP cache is empty, fetching IP
2021-08-19T10:02:21.533 DBG eventbus/event_bus.go:81                 > Published topic="consumer_ping_p2p" event={SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Duration:30.316983ms}
2021-08-19T10:02:26.567 DBG eventbus/event_bus.go:81                 > Published topic="consumer_ping_p2p" event={SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Duration:31.293013ms}
2021-08-19T10:02:29.157 DBG core/connection/manager.go:865           > Received p2p keepalive ping with SessionID=e2113530-d05f-4f5d-95aa-d99100e33606
2021-08-19T10:02:31.603 DBG eventbus/event_bus.go:81                 > Published topic="consumer_ping_p2p" event={SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Duration:31.681942ms}
2021-08-19T10:02:36.641 DBG eventbus/event_bus.go:81                 > Published topic="consumer_ping_p2p" event={SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Duration:33.373405ms}
2021-08-19T10:02:36.793 ERR core/location/cache.go:102               > Location update failed error="failed to execute request: Get https://testnet3-location.mysterium.network/api/v1/location: net/http: request canceled (Client.Timeout exceeded while awaiting headers)"
2021-08-19T10:02:36.795 DBG core/location/oracle_resolver.go:64      > Detecting with oracle resolver
2021-08-19T10:02:36.810 ERR core/ip/resolver.go:101                  > could not reach location service, will use fallbacks error="Get https://testnet3-location.mysterium.network/api/v1/location: net/http: request canceled (Client.Timeout exceeded while awaiting headers)"
2021-08-19T10:02:37.008 DBG core/connection/manager.go:391           > Sending session status P2P message to "p2p-session-connectivity-status": ConsumerID:"0xa53a9a07970ab2fdc04217e2d4a807bd350203eb"  SessionID:"e2113530-d05f-4f5d-95aa-d99100e33606"  Code:1000
2021-08-19T10:02:39.914 DBG config/config.go:224                     > Returning default value chain-id:80001
2021-08-19T10:02:39.914 DBG identity/selector/handler.go:98          > Attempting to use last identity
2021-08-19T10:02:39.915 DBG identity/selector/handler.go:103         > Found identity in cache: 0xa53a9a07970ab2fdc04217e2d4a807bd350203eb
2021-08-19T10:02:39.915 DBG identity/manager.go:153                  > Unlocked identity found in cache, skipping keystore: 0xa53a9a07970ab2fdc04217e2d4a807bd350203eb
2021-08-19T10:02:39.916 DBG identity/selector/handler.go:108         > Unlocked identity: 0xa53a9a07970ab2fdc04217e2d4a807bd350203eb
2021-08-19T10:02:39.917 DBG core/ip/cached_resolver.go:75            > Found cached public IP
2021-08-19T10:02:41.673 DBG eventbus/event_bus.go:81                 > Published topic="consumer_ping_p2p" event={SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Duration:29.898887ms}
2021-08-19T10:02:41.709 WRN communication/nats/connection_wrap.go:102 > NATS: disconnected
2021-08-19T10:02:43.192 DBG core/connection/manager.go:865           > Received p2p keepalive ping with SessionID=e2113530-d05f-4f5d-95aa-d99100e33606
2021-08-19T10:02:45.422 WRN communication/nats/connection_wrap.go:103 > NATS: reconnected
2021-08-19T10:02:46.704 DBG eventbus/event_bus.go:81                 > Published topic="consumer_ping_p2p" event={SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Duration:30.129524ms}
2021-08-19T10:02:51.742 DBG eventbus/event_bus.go:81                 > Published topic="consumer_ping_p2p" event={SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Duration:36.011422ms}
2021-08-19T10:02:56.789 DBG eventbus/event_bus.go:81                 > Published topic="consumer_ping_p2p" event={SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Duration:42.904391ms}
2021-08-19T10:02:56.805 DBG core/location/oracle_resolver.go:64      > Detecting with oracle resolver
2021-08-19T10:02:56.807 DBG core/ip/cached_resolver.go:75            > Found cached public IP
2021-08-19T10:02:57.227 DBG core/connection/manager.go:865           > Received p2p keepalive ping with SessionID=e2113530-d05f-4f5d-95aa-d99100e33606
2021-08-19T10:03:01.808 DBG core/ip/cached_resolver.go:75            > Found cached public IP
2021-08-19T10:03:01.820 DBG eventbus/event_bus.go:81                 > Published topic="consumer_ping_p2p" event={SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Duration:30.231897ms}
2021-08-19T10:03:06.812 DBG core/ip/cached_resolver.go:75            > Found cached public IP
2021-08-19T10:03:06.852 DBG eventbus/event_bus.go:81                 > Published topic="consumer_ping_p2p" event={SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Duration:29.995311ms}
2021-08-19T10:03:11.259 DBG core/connection/manager.go:865           > Received p2p keepalive ping with SessionID=e2113530-d05f-4f5d-95aa-d99100e33606
2021-08-19T10:03:11.816 DBG core/ip/cached_resolver.go:75            > Found cached public IP
2021-08-19T10:03:11.892 DBG eventbus/event_bus.go:81                 > Published topic="consumer_ping_p2p" event={SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Duration:36.162084ms}
2021-08-19T10:03:15.272 DBG session/pingpong/factory.go:136          > Received P2P message for "p2p-payment-invoice": AgreementID:"60933087532070207597448730368511591419058256564880633397550459210214220949217"  AgreementTotal:"8801332126515"  TransactorFee:"0"  Hashlock:"ea7435fccb052a2154e7a7533ad1f65336b1e726f671ab1b79775fca21c250ce"  Provider:"0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21"  ChainID:80001
2021-08-19T10:03:15.274 DBG session/pingpong/invoice_payer.go:146    > Invoice received: {60933087532070207597448730368511591419058256564880633397550459210214220949217 8801332126515 0 ea7435fccb052a2154e7a7533ad1f65336b1e726f671ab1b79775fca21c250ce 0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21 80001}
2021-08-19T10:03:15.275 DBG session/pingpong/price_calculator.go:51  > Calculated price 2177532996695476. Time component: 1.855597157720782856e+12, data component: 2.1756773995377569501e+15
2021-08-19T10:03:15.277 DBG session/pingpong/invoice_payer.go:191    > Estimated tolerance 1.612, upper bound 3510017401235066
2021-08-19T10:03:15.278 DBG session/pingpong/invoice_payer.go:244    > Loaded previous state: already promised: 5360043187212550
2021-08-19T10:03:15.280 DBG session/pingpong/invoice_payer.go:245    > Incrementing promised amount by 8801332126514
2021-08-19T10:03:15.282 DBG session/pingpong/exchange_messaging.go:68 > Sending P2P message to "p2p-payment-message": Promise:{ChannelID:"\\`\x14I\xa3~\xf8\xf8\xb5D\xe2\x04\x8e<:\x93A\xa4#\x12"  Amount:"5368844519339064"  Fee:"0"  Hashlock:"\xeat5\xfc\xcb\x05*!T\xe7\xa7S:\xd1\xf6S6\xb1\xe7&\xf6q\xab\x1byw_\xca!\xc2P\xce"  ChainID:80001  Signature:"C\xcc\x01\tw\xb4\x17\xf9\x8c\xfe\x809L\x0e\xb5\xdc\xfct\xda\x7f\x82\x88\xca建\xf4\xd4\xd3\x17\x8a\x9e\x1fnQA_\x92\x80\x02\x05\xcc7\xea\x18N\x9c\x7fqs\xfa8<\xa52\x1dxcM?\x1b\t\xb3\xa7\x1c"}  AgreementID:"60933087532070207597448730368511591419058256564880633397550459210214220949217"  AgreementTotal:"8801332126515"  Provider:"0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21"  Signature:"42ca92083b59882039f676e76c58d9bb3573b0aa17e5b6bc18ae209ea751f4921a4acf48d43391911d5d92f1737e1e4005da0240eb9d76e9ead0089b5b0b11a81c"  HermesID:"0x7119442C7E627438deb0ec59291e31378F88DD06"  ChainID:80001
2021-08-19T10:03:15.314 DBG eventbus/event_bus.go:81                 > Published topic="invoice_paid" event={ConsumerID:{Address:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb} SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Invoice:{AgreementID:+60933087532070207597448730368511591419058256564880633397550459210214220949217 AgreementTotal:+8801332126515 TransactorFee:+0 Hashlock:ea7435fccb052a2154e7a7533ad1f65336b1e726f671ab1b79775fca21c250ce Provider:0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21 ChainID:80001}}
2021-08-19T10:03:15.321 DBG consumer/session/session_storage.go:287  > Session e2113530-d05f-4f5d-95aa-d99100e33606 updated
2021-08-19T10:03:15.326 DBG eventbus/event_bus.go:81                 > Published topic="consumer_grand_total_change" event={Current:+5368844519339064 ChainID:80001 HermesID:0x7119442C7E627438deb0ec59291e31378F88DD06 ConsumerID:{Address:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb}}
2021-08-19T10:03:15.328 DBG config/config.go:224                     > Returning default value payments.consumer.offchain-expiration:30m0s
2021-08-19T10:03:15.329 DBG eventbus/event_bus.go:81                 > Published topic="balance_change" event={Identity:{Address:0xa53a9a07970ab2fdc04217e2d4a807bd350203eb} Previous:+99999976724922872496 Current:+99999926273744415694}
2021-08-19T10:03:15.524 DBG config/config.go:224                     > Returning default value default-currency:MYSTT
2021-08-19T10:03:15.526 DBG config/config.go:224                     > Returning default value default-currency:MYSTT
2021-08-19T10:03:15.527 INF core/state/state.go:482                  > Session ID: e2113530-d05f-4f5d-95aa-d99100e33606, state: Connected, duration: 1m8.303903105s data: 35.1KiB/30.5KiB, throughput: 0bs/0bs, spent: 0.000009MYSTT
2021-08-19T10:03:16.819 DBG core/ip/cached_resolver.go:75            > Found cached public IP
2021-08-19T10:03:16.924 DBG eventbus/event_bus.go:81                 > Published topic="consumer_ping_p2p" event={SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Duration:30.014793ms}
2021-08-19T10:03:21.821 DBG core/ip/cached_resolver.go:75            > Found cached public IP
2021-08-19T10:03:21.962 DBG eventbus/event_bus.go:81                 > Published topic="consumer_ping_p2p" event={SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Duration:31.544366ms}
2021-08-19T10:03:25.292 DBG core/connection/manager.go:865           > Received p2p keepalive ping with SessionID=e2113530-d05f-4f5d-95aa-d99100e33606
2021-08-19T10:03:26.823 DBG core/ip/cached_resolver.go:75            > Found cached public IP
2021-08-19T10:03:26.994 DBG eventbus/event_bus.go:81                 > Published topic="consumer_ping_p2p" event={SessionID:e2113530-d05f-4f5d-95aa-d99100e33606 Duration:29.954846ms}
```

**Comment by Waldz** [comment]
Same with provider: `0x965fa3f4e1184c0dcf6bd0ee36b975fa9b431e21`
Problem: Location check lasts 20s and fails: `server response invalid: 503 Service Unavailable (http://supernode-tunnel-1:4050/connection/location). Possible error: failed to execute request: Get https://testnet3-location.mysterium.network/api/v1/location: net/http: request canceled (Client.Timeout exceeded while awaiting headers)`

Logs:
```
2021-08-19 09:06:09 [WARN] [0x16bb9b72c54f57e8a0a1a4917b2241f18cd74a8e:supernode-tunnel-0] Location check failed. server response invalid: 503 Service Unavailable (http://supernode-tunnel-0:4050/connection/location). Possible error: failed to execute request: Get https://testnet3-location.mysterium.network/api/v1/location: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
```

**Comment by Snawoot** [comment]
> Problem: Location check lasts 20s and fails: `server response invalid: 503 Service Unavailable (http://supernode-tunnel-1:4050/connection/location). Possible error: failed to execute request: Get https://testnet3-location.mysterium.network/api/v1/location: net/http: request canceled (Client.Timeout exceeded while awaiting headers)`

Caught this problem too, but after a while it disappeared. It seems built-in golang HTTP client wasn't able to reach that resource. While `curl` was working fine in same time.

It is possible that `http.Transport` used by `http.Client` had established HTTP session before VPN connection was established and reused it in keepalive mode, so it appeared hang. If so, it's pretty easy to fix adjusting transport settings or killing all idle connections of transport on VPN state change. But definitely this issue is not specific to testnet3.

**Comment by Snawoot** [comment]
Continuing to look into `0x57daba353514f004c86edf2e3892740ec4d98c4d` problem...

**Comment by Snawoot** [comment]
`0x57daba353514f004c86edf2e3892740ec4d98c4d` is tricky one. I've manually forwarded ports on my consumer and still got similar errors.

To debug this I decided to check if port on Provider is really open. Trying to connect and watching logs:

```
2021-08-19T14:54:14.987 INF firewall/outgoing_firewall_noop.go:48 > Allow IP 86.134.59.214 access
2021-08-19T14:54:14.987 DBG p2p/dialer.go:282     > Skipping provider ping
2021-08-19T14:54:14.999 DBG p2p/dialer.go:133     > Received handlers ready message from provider
2021-08-19T14:54:15.001 DBG p2p/channel.go:212    > Creating p2p channel with local addr: 192.168.1.62:52059, UDP session addr: 127.0.0.1:49600, proxy addr: 127.0.0.1:44326, remote peer addr: x.x.x.x:51926
```

In parallel console, quickly issuing command and trying to send some data:

```
dt1:~ # nc -u 86.134.59.214 51926
aa
dt1:~ # 
```

netcat terminated instantly, because it received ICMP "port closed" message. Judging by port ranges provider has, it is likely on manual port range, but ports are not actually forwarded on router. So it's plain misconfiguration.

I do not see here a case related to testnet3, so I'm closing this issue and opening new one about disabling HTTP connection keep-alive for Go http client.

**Comment by Waldz** [comment]
> #3835
This is nice to nice to have, but not critical. This issue extends connection creation to >30s, because my connect flow is:
- location check ?s
- connection create ~10s
- location check after connection 20s (and yes, it's not 100% reproducible)

> I do not see here a case related to testnet3.
What worries me is this one, it was not addressed for long time. And misconfigured providers will impact a lack of residential  for us

**Comment by Waldz** [comment]
Testing residential providers on Testnet3:
```
GB 2/3 are connectable
US 5/5 are connectable
IT 2/2 are connectable
```
